sql-plugin/src/main/scala/com/nvidia/spark/rapids/SparkShims.scala

```diff
@@ -221,6 +221,12 @@ trait SparkShims {
    */
   def shouldFallbackOnAnsiTimestamp(): Boolean
 
+  /**
+   * This is to support ANSI mode: optionally return null result if element not exists
+   * in array/map.
+   */
+  def shouldFailOnElementNotExists(): Boolean = false
```

Comment:
So by default it does not fail, even if ANSI is enabled? 

Is the following correct for what we want?
|version | throw exception if key not found...|
|-|-|
|3.0.*| Never|
|3.1.*| If ANSI is enabled |
|3.2.*| If ANSI is enabled |
|3.3.* | If ANSI is enabled and strictIndexOperator is enabled (SQLConf includes both together in the helper method) |

Reply:
The table is correct.

`strictIndexOperator` is true by default